File: ovvium-pos/src/app/model/response/OrderResponse.ts
import { ProductResponse } from './ProductResponse';
import { ResourceIdResponse } from './ResourceIdResponse';
import { UserResponse } from './UserResponse';
import { OrderGroupChoiceResponse } from 'app/model/response/OrderGroupChoiceResponse';

export class OrderResponse extends ResourceIdResponse {
    user: UserResponse;
    product: ProductResponse;
    paymentStatus: string;
    issueStatus: string;
    price: number;
    tax: number;
    serviceTime: string;
    orderTime: number;
    notes?: string;
    groupChoices: Array<OrderGroupChoiceResponse>;

    constructor(orderResponse: OrderResponse = {} as OrderResponse) {
        super(orderResponse.id);
        this.user = orderResponse.user;
        this.product = orderResponse.product;
        this.paymentStatus = orderResponse.paymentStatus;
        this.issueStatus = orderResponse.issueStatus;
        this.price = orderResponse.price;
        this.tax = orderResponse.tax;
        this.serviceTime = orderResponse.serviceTime;
        this.orderTime = orderResponse.orderTime;
        this.notes = orderResponse.notes;
        this.groupChoices = orderResponse.groupChoices;
    }
}

File: ovvium-backend/src/main/java/com/ovvium/services/web/controller/bff/v1/transfer/response/order/OrderResponse.java
package com.ovvium.services.web.controller.bff.v1.transfer.response.order;

import com.ovvium.services.model.bill.IssueStatus;
import com.ovvium.services.model.bill.Order;
import com.ovvium.services.model.bill.PaymentStatus;
import com.ovvium.services.transfer.response.order.SelectedProductOptionResponse;
import com.ovvium.services.web.controller.bff.v1.transfer.response.common.ResourceIdResponse;
import com.ovvium.services.web.controller.bff.v1.transfer.response.product.ProductResponse;
import com.ovvium.services.web.controller.bff.v1.transfer.response.user.UserResponse;
import lombok.Getter;

import java.util.List;
import java.util.stream.Collectors;

@Getter
public final class OrderResponse extends ResourceIdResponse {

    private final UserResponse user;
    private final ProductResponse product;
    private final IssueStatus issueStatus;
    private final PaymentStatus paymentStatus;
    private final double basePrice;
    private final double price;
    private final double tax;
    private final String serviceTime;
    private final long orderTime;
    private final String notes;
    private final List<OrderGroupChoiceResponse> groupChoices;
    private final List<SelectedProductOptionResponse> selectedOptions;

    public OrderResponse(Order order, UserResponse userResponse, ProductResponse productResponse, List<OrderGroupChoiceResponse> groupChoices) {
        super(order);
        this.issueStatus = order.getIssueStatus();
        this.paymentStatus = order.getPaymentStatus();
        this.user = userResponse;
        this.product = productResponse;
        this.price = order.getPrice().asDouble();
        this.basePrice = order.getBasePrice().asDouble();
        this.tax = order.getTax();
        this.serviceTime = order.getServiceTime().name();
        this.orderTime = order.getCreated().toEpochMilli();
        this.notes = order.getNotes().orElse(null);
        this.groupChoices = groupChoices;
        this.selectedOptions = order.getSelectedOptions().stream()
                .map(SelectedProductOptionResponse::new)
                .collect(Collectors.toList());

    }

}
